pkg/node/manager/manager.go

```diff
@@ -565,9 +567,17 @@ func (m *Manager) upsertIntoIDMD(prefix string, id identity.NumericIdentity) {
 	}
 }
 
+func (m *Manager) removeFromIDMD(prefix string, id identity.NumericIdentity, src source.Source) {
+	if id == identity.ReservedIdentityHost {
+		m.ipcache.RemoveMetadata(prefix, labels.LabelHost, src)
+	} else {
+		m.ipcache.RemoveMetadata(prefix, labels.LabelRemoteNode, src)
```

Comment:
Looks like this would be better as a switch statement, as the default here is assuming that the ID is `RemoteNode`

Reply:
Yeah I'd be inclined to agree. I did it this way to match the pattern in the existing `upsertIntoIDMD()`. Moving forward in 1.14 or later this path won't be around so this stylistic change is neither here nor there to me. Might be a little simpler to reason through.

Reply:
Are there any negative side effects if the label passed here does not exist for the IP?

Reply:
The inner implementation of `removeLabels()` will do a lookup for the prefix and if nothing exists, return nil without complaint. If the prefix has labels but the label does not exist, then `removeLabels()` will use `Labels.Remove()` to remove the label from the set, which is a no-op and that function looks like it handles removing non-existing labels with no trouble. There are a few error check cases where if internal state is a bit inconsistent (for instance, there's no corresponding identity for the target labels for the prefix) then Cilium will emit warnings, which is not very actionable but it won't do anything crazy like crash. This could potentially lead to releasing a reference on an existing identity and removing the entry from the ipcache even though the change in labels should mean this is a no-op, so in that case there is a chance that it could interrupt connectivity for the IP corresponding to the node. So in the general sense this code has not been written defensively to prevent this particular case. At the same time, the caller of this function here will just directly do an `m.ipcache.Delete()` immediately after this, so practically speaking the resulting agent+datapath state in that case is exactly the same whether the label previously existed in the metadata cache or not.

So TL;DR, no there should not be any (additional) negative side effects if there was such a bug.